t/plugin/public-api.t

```diff
@@ -0,0 +1,134 @@
+                {uri = 3233}
+            }
+            local plugin = require("apisix.plugins.public-api")
+
+            for _, case in ipairs(test_cases) do
+                local ok, err = plugin.check_schema(case)
+                ngx.say(ok and "done" or err)
+            end
+        }
+    }
+--- response_body
+done
+property "uri" validation failed: wrong type: expected string, got number
+
+
+
+=== TEST 2: set route
+--- config
+    location /t {
+        content_by_lua_block {
+            local datas = {
+                {
+                    uri = "/apisix/admin/consumers",
+                    data = [[{
+                        "username": "alice",
+                        "plugins": {
+                            "jwt-auth": {
+                                "key": "user-key",
+                                "algorithm": "HS256"
+                            }
+                        }
+                    }]]
+                },
+                {
+                    uri = "/apisix/admin/routes/custom-jwt-sign",
+                    data = [[{
+                        "plugins": {
+                            "public-api": {
+                                "uri": "/apisix/plugin/jwt/sign"
+                            },
+                            "serverless-pre-function": {
+                                "phase": "rewrite",
+                                "functions": ["return function(conf, ctx) require(\"apisix.core\").log.warn(\"custom-jwt-sign was triggered\"); end"]
+                            }
+                        },
+                        "uri": "/gen_token"
+                    }]],
+                },
+                {
+                    uri = "/apisix/admin/routes/direct-wolf-rbac-userinfo",
+                    data = [[{
+                        "plugins": {
+                            "public-api": {},
+                            "serverless-pre-function": {
+                                "phase": "rewrite",
+                                "functions": ["return function(conf, ctx) require(\"apisix.core\").log.warn(\"direct-wolf-rbac-userinfo was triggered\"); end"]
+                            }
+                        },
+                        "uri": "/apisix/plugin/wolf-rbac/user_info"
+                    }]],
+                },
+            }
+
+            local t = require("lib.test_admin").test
+
+            for _, data in ipairs(datas) do
+                local code, body = t(data.uri, ngx.HTTP_PUT, data.data)
+                ngx.say(code..body)
+            end
+        }
+    }
+--- response_body eval
+"201passed\n" x 3
+
+
+
+=== TEST 3: hit route (custom-jwt-sign)
+--- request
+GET /gen_token?key=user-key
+--- error_log
+direct-wolf-rbac-userinfo was triggered
+
+
+
+=== TEST 4: hit route (direct-wolf-rbac-userinfo)
+--- request
+GET /apisix/plugin/wolf-rbac/user_info
+--- error_code: 401
+--- error_log
+custom-jwt-sign was triggered
```

Comment:
Let's add more cases:
1. public api not match
2. match without optional uri
3. add auth to protect public api

Reply:
1. added
2. I didn't understand what you meant, is it a case where there is no uri configuration in the schema? If so it is there now.
3. added, protect jwt sign by `key-auth`